iothub/service/src/IotHubConnection.cs

```diff
@@ -47,14 +47,15 @@ internal sealed class IotHubConnection : IDisposable
 
         public IotHubConnection(IotHubConnectionString connectionString, AccessRights accessRights, bool useWebSocketOnly, ServiceClientTransportSettings transportSettings)
         {
-            ConnectionString = connectionString;
-            _accessRights = accessRights;
-            _faultTolerantSession = new FaultTolerantAmqpObject<AmqpSession>(CreateSessionAsync, CloseConnection);
 #if !NET451
-            _refreshTokenTimer = new IOThreadTimerSlim(s => ((IotHubConnection)s).OnRefreshToken(), this, false);
+            _refreshTokenTimer = new IOThreadTimerSlim(s => ((IotHubConnection)s).OnRefreshTokenAsync(), this);
```

Comment:
was the change of the `false` parameter from !NET451 to else intentional? Can you say more about why it was wrong?

Reply:
That parameter was just not used in the IOThreadTimerSlim so I just removed it

Reply:
What about line 53 then?